frontend/src/components/MapView/MapTooltip/PopupCharts.tsx

```diff
@@ -0,0 +1,246 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExternalLinkAlt, faTimes } from '@fortawesome/free-solid-svg-icons';
+import {
+  Button,
+  IconButton,
+  WithStyles,
+  createStyles,
+  withStyles,
+} from '@material-ui/core';
+import { MapTooltipState } from 'context/tooltipStateSlice';
+import i18n, { isEnglishLanguageSelected } from 'i18n';
+
+import ChartSection from '../LeftPanel/ChartsPanel/ChartSection';
+import { oneYearInMs } from '../LeftPanel/utils';
+import DownloadCsvButton from '../DownloadCsvButton';
+import { buildCsvFileName } from '../utils';
+
+const chartLayers = getWMSLayersWithChart();
+const { countryAdmin0Id, country, multiCountry } = appConfig;
+const MAX_ADMIN_LEVEL = appConfig.multiCountry ? 3 : 2;
+const admin0Offset = multiCountry ? 0 : 1;
+const boundaryLayer = getBoundaryLayersByAdminLevel(MAX_ADMIN_LEVEL);
+
+const getProperties = (
+  layerData: LayerData<BoundaryLayerProps>['data'],
+  name: string,
+) => {
+  const features = layerData.features.filter(
+    elem => elem.properties && elem.properties[name],
+  );
+
+  if (!features) {
+    return null;
+  }
+  return features[0].properties;
+};
+
+const styles = () =>
+  createStyles({
+    selectChartContainer: {
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'start',
+    },
+    selectLevelButton: {
+      textTransform: 'none',
+    },
+    selectLevelButtonValue: {
+      display: 'flex',
+      gap: '8px',
+      alignItems: 'center',
+      justifyContent: 'start',
+    },
+    selectLevelButtonText: {
+      overflow: 'hidden',
+      whiteSpace: 'nowrap',
+      textOverflow: 'ellipsis',
+      maxWidth: '280px',
+    },
+    closeButton: {
+      color: 'white',
+      position: 'absolute',
+      right: 0,
+      top: 0,
+    },
+    chartsContainer: {
+      position: 'relative',
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    charts: {
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    chartContainer: {
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    chartSection: {
+      height: '240px',
+      width: '400px',
+      flexGrow: 1,
+    },
+  });
+
+interface PopupChartProps extends WithStyles<typeof styles> {
+  popup: MapTooltipState;
+  setPopupTitle: React.Dispatch<React.SetStateAction<string>>;
+}
+type AdminLevel = 0 | 1 | 2;
+
+const PopupChart = ({ popup, setPopupTitle, classes }: PopupChartProps) => {
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+  const { data } = boundaryLayerData || {};
+  const mapState = useSelector(layersSelector);
+  const dataForCsv = useRef<{ [key: string]: any[] }>({});
+  const [adminLevel, setAdminLevel] = useState<AdminLevel | undefined>(
+    undefined,
+  );
+
+  const adminLevelsNames = useCallback(() => {
```

Comment:
The whole logic around admin level names and admin level selection is quite confusing and error prone.

- It took me a while to fix the code and make it work well for multi-country deployments without breaking other deploys.

Don't we have a better way than splitting the locationName strings to get the admin names?

Reply:
Based on what [I did in #974](https://github.com/WFP-VAM/prism-app/pull/974/files#diff-cb442affc6c2fb89fbf76cbc5bc40b49d94d3d6e9978824ac6454edb3ce5b2a0R182) (which is itself copy/pasted from the previous iteration of code), I think you could just do:
```javascript
const splitNames = isEnglishLanguageSelected(i18n)
      ? boundaryLayer.locationName
      : boundaryLayer.locationLocalName;
```

We seem to have code like this copy/pasted across a number of components.